### PR TITLE
TD-API: Add the FetchFederationRelationship method to the datastore

### LINKED ISSUE #2504: Federation Management API: Implement DataStore support for GetFederationRelationship
The implementation should be a fairly simple CRUD operation.

### PR DESCRIPTION
**Pull request by marcosy**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

Datastore

**Description of change**

Add the FetchFederationRelationship() method to the datastore.

**Which issue this PR fixes**

Fixes #2504

Signed-off-by: Marcos Yedro <marcosyedro@gmail.com>

### CODE REVIEW

File: pkg/common/telemetry/server/datastore/wrapper.go [hunk 1]
```diff
@@ -111,6 +112,12 @@ func (w metricsWrapper) FetchRegistrationEntry(ctx context.Context, entryID stri
 	return w.ds.FetchRegistrationEntry(ctx, entryID)
 }
 
+func (w metricsWrapper) FetchFederationRelationship(ctx context.Context, trustdomain spiffeid.TrustDomain) (_ *datastore.FederationRelationship, err error) {
```
Comment (azdagron) [inline pkg/common/telemetry/server/datastore/wrapper.go:115]: nitpick: `td` or `trustDomain` for TrustDomain variable names

File: pkg/server/datastore/sqlstore/sqlstore.go [hunk 2]
```diff
@@ -3265,6 +3284,48 @@ func createFederationRelationship(tx *gorm.DB, fr *datastore.FederationRelations
...
+	switch fr.BundleEndpointProfile {
+	case datastore.BundleEndpointWeb:
+	case datastore.BundleEndpointSPIFFE:
+		endpointSPIFFEID, err := spiffeid.FromString(model.EndpointSPIFFEID)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse bundle endpoint SPIFFE ID: %w", err)
+		}
+		fr.EndpointSPIFFEID = endpointSPIFFEID
+
+		bundle, err := fetchBundle(tx, trustdomain.IDString())
```
Comment (azdagron) [inline pkg/server/datastore/sqlstore/sqlstore.go:3317]: @amartinezfayo , @evan2645, I'm curious what you think of this. Do we expect folks who configure a relationship with a SPIFFE profile to see the bundle they configured or the latest bundle when they fetch?

**Review by azdagron — COMMENTED** [review]
Looks great, @marcosy. I think we are ok as-is, but I want to make sure @amartinezfayo and @evan2645 are ok with returning the latest bundle.

File: pkg/server/datastore/sqlstore/sqlstore.go (hunk 2 above)
Comment (evan2645) [reply pkg/server/datastore/sqlstore/sqlstore.go:3317]: I _feel_ like the natural thing is for this to be the latest bundle? Especially if folks want to copy/paste or move things around to other clusters and have it work.

It was also my understanding that when a relationship is created and a bundle is explicitly supplied, that we would simply write this bundle to the bundles table and use it to bootstrap as we normally would (as opposed to storing the originally configured bundle with the relationship). If we were doing that, I think the "latest" bundle is all we could afford anyway?

Comment (azdagron) [reply pkg/server/datastore/sqlstore/sqlstore.go:3317]: You're right. I concur. Ship it!

File: pkg/server/datastore/sqlstore/sqlstore.go [hunk 3]
```diff
@@ -468,6 +470,23 @@ func (ds *Plugin) CreateFederationRelationship(ctx context.Context, fr *datastor
 	})
 }
 
+// FetchFederationRelationship fetches the federation relationship that matches
+// the given trust domain. If the federation relationship is not found, nil is returned.
+func (ds *Plugin) FetchFederationRelationship(ctx context.Context, trustDomain spiffeid.TrustDomain) (fr *datastore.FederationRelationship, err error) {
+	if trustDomain.IsZero() {
+		return nil, status.Errorf(codes.InvalidArgument, "trust domain is required")
```
Comment (amartinezfayo) [inline pkg/server/datastore/sqlstore/sqlstore.go:477]: nit: no need for formatted error
```suggestion
		return nil, status.Error(codes.InvalidArgument, "trust domain is required")
```

**Review by azdagron — DISMISSED** [review]
\o/, thanks @marcosy !

File: pkg/server/datastore/sqlstore/sqlstore.go (hunk 2 above)
Comment (amartinezfayo) [reply pkg/server/datastore/sqlstore/sqlstore.go]: Yes, I also think that he natural behavior is to be the latest bundle.

**Review by azdagron — APPROVED** [review]

**Pull request merged**